apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/preadj/AbstractR4ResourceProviderTest.java

```diff
@@ -0,0 +1,314 @@
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import ca.uhn.fhir.rest.annotation.IdParam;
+import ca.uhn.fhir.rest.annotation.Read;
+import ca.uhn.fhir.rest.api.server.RequestDetails;
+import ca.uhn.fhir.rest.server.exceptions.ResourceNotFoundException;
+import com.codahale.metrics.MetricRegistry;
+import gov.cms.bfd.model.rda.PreAdjFissClaim;
+import gov.cms.bfd.server.war.r4.providers.preadj.common.ClaimDao;
+import gov.cms.bfd.server.war.r4.providers.preadj.common.ResourceTransformer;
+import gov.cms.bfd.server.war.r4.providers.preadj.common.ResourceTypeV2;
+import gov.cms.bfd.server.war.utils.AssertUtils;
+import gov.cms.bfd.server.war.utils.ReflectionTestUtils;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.util.Optional;
+import javax.annotation.PostConstruct;
+import javax.persistence.EntityManager;
+import org.hl7.fhir.r4.model.Claim;
+import org.hl7.fhir.r4.model.IdType;
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class AbstractR4ResourceProviderTest {
+
+  @Test
+  public void shouldHaveInitMethodAnnotatedWithPostConstruct() throws NoSuchMethodException {
+    Method method = AbstractR4ResourceProvider.class.getDeclaredMethod("init");
+
+    assertTrue(method.isAnnotationPresent(PostConstruct.class));
+  }
+
+  @Test
+  public void shouldCreateDaoInstance() throws NoSuchFieldException, IllegalAccessException {
+    AbstractR4ResourceProvider<?> providerSpy = spy(AbstractR4ResourceProvider.class);
+
+    EntityManager mockEntityManager = mock(EntityManager.class);
+    MetricRegistry mockMetricRegistry = mock(MetricRegistry.class);
+
+    ReflectionTestUtils.setField(providerSpy, "entityManager", mockEntityManager);
+    ReflectionTestUtils.setField(providerSpy, "metricRegistry", mockMetricRegistry);
+
+    doNothing().when(providerSpy).setResourceType();
+
+    providerSpy.init();
+
+    ClaimDao expected = new ClaimDao(mockEntityManager, mockMetricRegistry);
+    Object actual = ReflectionTestUtils.getField(providerSpy, "claimDao");
+
+    assertEquals(expected, actual);
+  }
+
+  @Test
+  public void shouldInvokeSetResourceType() throws NoSuchFieldException, IllegalAccessException {
+    AbstractR4ResourceProvider<?> providerSpy = spy(AbstractR4ResourceProvider.class);
+
+    EntityManager mockEntityManager = mock(EntityManager.class);
+    MetricRegistry mockMetricRegistry = mock(MetricRegistry.class);
+
+    ReflectionTestUtils.setField(providerSpy, "entityManager", mockEntityManager);
+    ReflectionTestUtils.setField(providerSpy, "metricRegistry", mockMetricRegistry);
+
+    doNothing().when(providerSpy).setResourceType();
+
+    providerSpy.init();
+
+    verify(providerSpy, times(1)).init();
```

Comment:
Is this verifying the right call?  Shouldn't it be verifying `setResourceType()`?